File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function (matrix, target) {
    const n = matrix.length;
    const m = matrix[0].length;
    let high = 0;
    let low = m - 1;
    while (low >= 0 && high <= n - 1) {
        if (matrix[high][low] == target) {
            return true
        }
        else if (matrix[high][low] < target) {
            high++;
        }
        else {
            low--;
        }

    }
    return false

};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size();    // 5
        int n = matrix[0].size(); // 4
        int low = 0;
        int high = n- 1;

        while (high >= 0 && low <= m - 1) {
            if (matrix[low][high] == target) {
                return true;
            } else if (matrix[low][high] < target) {
                low++;
            } else {
                high--;
            }
        }
        return false;
    }
};